File: src/main/java/com/leijendary/spring/boot/template/config/PropertiesConfiguration.java
package com.leijendary.spring.boot.template.config;

import com.leijendary.spring.boot.template.config.properties.AuthProperties;
import com.leijendary.spring.boot.template.config.properties.CorsProperties;
import com.leijendary.spring.boot.template.config.properties.InfoProperties;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties({
        AuthProperties.class,
        CorsProperties.class,
        InfoProperties.class
})
public class PropertiesConfiguration {
}

File: src/main/kotlin/com/leijendary/spring/template/core/config/PropertiesConfiguration.kt
package com.leijendary.spring.template.core.config

import com.leijendary.spring.template.core.config.properties.*
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(
    AuthProperties::class,
    AwsS3Properties::class,
    DataSourcePrimaryProperties::class,
    DataSourceReadonlyProperties::class,
    EmissionProperties::class,
    InfoProperties::class,
    NumberProperties::class,
    RetryProperties::class
)
class PropertiesConfiguration 
